src/test/java/uk/gov/companieshouse/model/validator/AppealReasonValidatorTest.java

```diff
@@ -0,0 +1,76 @@
+package uk.gov.companieshouse.model.validator;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.assertj.core.util.Lists;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.companieshouse.TestData;
+import uk.gov.companieshouse.exception.AppealReasonException;
+import uk.gov.companieshouse.model.Attachment;
+import uk.gov.companieshouse.model.IllnessReason;
+import uk.gov.companieshouse.model.OtherReason;
+import uk.gov.companieshouse.model.Reason;
+
+@ExtendWith(SpringExtension.class)
+class AppealReasonValidatorTest {
+
+    @Mock
+    private Reason mockReason = new Reason();
+
+    @InjectMocks
+    private AppealReasonValidator appealReasonValidator;
+
+    @Test
+    void shouldThrowAnExceptionWhenItsNotIllnessOrOtherReason() {
+        assertThrows(AppealReasonException.class, () -> appealReasonValidator.validate(mockReason));
+    }
+
+    @Test
+    void shouldThrowAnExceptionWhenHasBothIllnessAndOtherReason() {
+        mockReason = new Reason(createOtherReason(), createIllnessReason());
+        assertThrows(AppealReasonException.class, () -> appealReasonValidator.validate(mockReason));
+    }
+
+    @Test
+    void shouldNotThrowErrorWhenItsIllnessReason(){
+        mockReason = new Reason(null, createIllnessReason());
+        assertDoesNotThrow(() -> appealReasonValidator.validate(mockReason));
+    }
+
+    @Test
+    void shouldNotThrowErrorWhenItsOtherReason(){
+        mockReason = new Reason(createOtherReason(), null);
+        assertDoesNotThrow(() -> appealReasonValidator.validate(mockReason));
+    }
+
+    private IllnessReason createIllnessReason(){
```

Comment:
Is this duplicated from the other place? It might be worth adding a test util class that handles constructing these.

Reply:
Done, Thanks!